lnd.go

```diff
@@ -666,46 +667,58 @@ func bakeMacaroon(ctx context.Context, svc *macaroons.Service,
 	return mac.M().MarshalBinary()
 }
 
-// genMacaroons generates three macaroon files; one admin-level, one for
-// invoice access and one read-only. These can also be used to generate more
-// granular macaroons.
-func genMacaroons(ctx context.Context, svc *macaroons.Service,
-	admFile, roFile, invoiceFile string) error {
+// saveMacaroon bakes a macaroon with the specified macaroon permissions and
+// writes it to a file with the given filename and file permissions.
+func saveMacaroon(ctx context.Context, svc *macaroons.Service, filename string,
+	macaroonPermissions []bakery.Op, filePermissions fs.FileMode) error {
 
-	// First, we'll generate a macaroon that only allows the caller to
-	// access invoice related calls. This is useful for merchants and other
-	// services to allow an isolated instance that can only query and
-	// modify invoices.
-	invoiceMacBytes, err := bakeMacaroon(ctx, svc, invoicePermissions)
+	macaroonBytes, err := bakeMacaroon(ctx, svc, macaroonPermissions)
 	if err != nil {
 		return err
 	}
-	err = ioutil.WriteFile(invoiceFile, invoiceMacBytes, 0644)
+	err = ioutil.WriteFile(filename, macaroonBytes, filePermissions)
```

Comment:
While we're at it, we should use `os.WriteFile()` since the `ioutil` is deprecated.